Manage pending changes to RW accounts

andre8244 opened the pull request:
Perform changes to RoadWarrior accounts without restarting the service immediately

https://github.com/NethServer/dev/issues/6177

andre8244 committed:
api & ui: manage pending changes to RW accounts

Perform changes to RoadWarrior accounts without restarting the
service immediately

andre8244 commented:
Release after https://github.com/NethServer/nethserver-openvpn/pull/68

edospadoni commented on api/openvpn-rw/read:
```diff
@@ -259,6 +259,14 @@ iOS: https://itunes.apple.com/it/app/openvpn-connect/id590379981
+    if (-e '/var/spool/openvpn/road-warrior-lock') {
+        $pendingChanges = 1;
+    }
+    $ret->{'pendingChanges'} = $pendingChanges;
```
We can improve with this:
```perl
$ret->{'pendingChanges'} = (-e '/var/spool/openvpn/road-warrior-lock') ? 1 : 0;
```

@gsanchietti is the perl magician, maybe he can confirm

gsanchietti replied:
I agree, you can even remove the declaration on line 264: 8 bytes of RAM spared! :D

edospadoni commented on ui/src/views/OpenVPNRW.vue:
```diff
@@ -22,6 +22,51 @@
+        {{$t('openvpn_rw.apply_changes')}}
+      </button>
+      <span :class="['pficon', 'pficon-warning-triangle-o']"></span>
+      <strong>{{$t('warning')}}.</strong>
```
It's a matter of taste, but we always use `:` after `Warning`. We change it to:
```js
<strong>{{$t('warning')}}:</strong>
```

andre8244 replied:
Ok. I actually copied it from the firewall module: https://github.com/NethServer/nethserver-firewall-base/blob/master/ui/src/App.vue#L131

edospadoni commented on ui/src/views/OpenVPNRW.vue:
```diff
@@ -22,6 +22,51 @@
+            <div class="modal-body">
+              <div class="alert alert-warning alert-dismissable">
+                <span class="pficon pficon-warning-triangle-o"></span>
+                <strong>{{$t('warning')}}.</strong>
```
It's a matter of taste, but we always use `:` after `Warning`. We change it to:
```js
<strong>{{$t('warning')}}:</strong>
```

andre8244 committed:
Incorporated nethserver-vpn-ui-save event

nethserver-vpn-ui-save event has been incorporated into nethserver-openvpn-save

andre8244 committed:
api & ui: minor fixes

andre8244 committed:
ui: UI fixes to RoadWarrior page
